packages/grid/x-data-grid/src/internals/hooks/features/columns/gridColumnsUtils.ts

```diff
@@ -31,6 +31,130 @@ export const computeColumnTypes = (customColumnTypes: GridColumnTypesRecord = {}
   return mergedColumnTypes;
 };
 
+/**
+ * Computes width for flex columns.
+ * Based on CSS Flexbox specification:
+ * https://drafts.csswg.org/css-flexbox-1/#resolve-flexible-lengths
+ */
+export function computeFlexColumnsWidth({
+  initialFreeSpace,
+  totalFlexUnits,
+  flexColumns,
+}: {
+  initialFreeSpace: number;
+  totalFlexUnits: number;
+  flexColumns: {
+    field: GridColDef['field'];
+    flex?: number;
+    minWidth?: number;
+    maxWidth?: number;
+  }[];
+}) {
+  const flexColumnsLookup: {
+    all: Record<
+      GridColDef['field'],
+      {
+        flex: number;
+        computedWidth: number;
+        frozen: boolean;
+      }
+    >;
+    frozenFields: GridColDef['field'][];
+    freeze: (field: GridColDef['field']) => void;
+  } = {
+    all: {},
+    frozenFields: [],
+    freeze: function freeze(field: GridColDef['field']) {
```

Comment:
I think we can avoid using `this`, which can always be a source of confusion when you don't know the code well.